include/envoy/network/address.h

```diff
@@ -125,5 +125,104 @@ class Instance {
 
 typedef std::shared_ptr<const Instance> InstancePtr;
 
+/*
+ * Parse an internet host address (IPv4 or IPv6) and create an Instance from it.
+ * The address must not include a port number.
+ * @param ipAddr string to be parsed as an internet address.
+ * @return pointer to the Instance, or nullptr if unable to parse the address.
+ */
+InstancePtr parseInternetAddress(const std::string& ipAddr);
```

Comment:
I don't mind the choice of namespace. I am a bit surprised about putting only abstract interfaces into include, though I see that is the general practice with only a few exceptions such as in include/envoy/http/header_map.h. Before I ask picky questions, is there a doc that lays out the philosophy for what goes where?